genome_designer/genome_finish/assembly.py

```diff
@@ -1,169 +1,379 @@
-                Dataset.TYPE.BWA_ALIGN).get_absolute_location()
+    velvet_opts = DEFAULT_VELVET_OPTS
+
+    # Find insertion metrics
+    ins_length, ins_length_sd = get_insert_size_mean_and_stdev(
+            sample_alignment)
+    velvet_opts['velvetg']['ins_length'] = ins_length
+    velvet_opts['velvetg']['ins_length_sd'] = ins_length_sd
 
-    alignment_prefix = os.path.join(data_dir, 'bwa_align')
+    # # Find expected coverage
+    # exp_cov = kmer_coverage(180, 100, 21)
+    # print exp_cov
+    # velvet_opts['velvetg']['exp_cov'] = exp_cov
 
-    # Extract SV indicating reads
-    unmapped_reads = alignment_prefix + '.unmapped.bam'
-    get_unmapped_reads(alignment_sorted, unmapped_reads)
+    for shallow_key in ['velveth', 'velvetg']:
+        if shallow_key in input_velvet_opts:
+            for deep_key in input_velvet_opts[shallow_key]:
+                velvet_opts[shallow_key][deep_key] = (
+                        input_velvet_opts[shallow_key][deep_key])
 
-    split_reads = alignment_prefix + '.split.bam'
-    get_split_reads(alignment_sorted, split_reads)
+    # Perform velvet assembly
+    contig_files = assemble_with_velvet(
+            data_dir, velvet_opts, sv_indicants_bam,
+            sample_alignment)
+
+    return contig_files
 
-    clipped_reads = alignment_prefix + '.clipped.bam'
-    get_clipped_reads(alignment_sorted, clipped_reads)
+
+def get_sv_indicating_reads(sample_alignment, input_sv_indicant_classes={},
+        overwrite=False):
+
+    sv_indicant_keys = [
+            Dataset.TYPE.BWA_PILED,
+            Dataset.TYPE.BWA_CLIPPED,
+            Dataset.TYPE.BWA_SPLIT,
+            Dataset.TYPE.BWA_UNMAPPED,
+            Dataset.TYPE.BWA_DISCORDANT
+    ]
+
+    sv_indicant_class_to_filename_suffix = {
+            Dataset.TYPE.BWA_PILED: 'piled',
+            Dataset.TYPE.BWA_CLIPPED: 'clipped',
+            Dataset.TYPE.BWA_SPLIT: 'split',
+            Dataset.TYPE.BWA_UNMAPPED: 'unmapped',
+            Dataset.TYPE.BWA_DISCORDANT: 'discordant'
+    }
+
+    sv_indicant_class_to_generator = {
+            Dataset.TYPE.BWA_PILED: get_piled_reads,
+            Dataset.TYPE.BWA_CLIPPED: get_clipped_reads_smart,
+            Dataset.TYPE.BWA_UNMAPPED: get_unmapped_reads
+    }
+
+    default_sv_indicant_classes = {
+            Dataset.TYPE.BWA_PILED: True,
+            Dataset.TYPE.BWA_CLIPPED: True,
+            Dataset.TYPE.BWA_SPLIT: True,
+            Dataset.TYPE.BWA_UNMAPPED: True,
+            Dataset.TYPE.BWA_DISCORDANT: True
+    }
+    default_sv_indicant_classes.update(input_sv_indicant_classes)
+
+    # Grab alignment bam file-path
+    alignment_bam = get_dataset_with_type(
+            sample_alignment,
+            Dataset.TYPE.BWA_ALIGN).get_absolute_location()
+
+    # HACK: Filter out unpaired mates
+    alignment_no_unpaired_bam = (os.path.splitext(alignment_bam)[0] +
+            '.no_unpaired.bam')
+    if not os.path.exists(alignment_no_unpaired_bam):
```

Comment:
Yes, because contigs are generated from an alignment.

For a user to get to the Contigs tab, he or she would have to select a reference genome and an alignment.